json_serializable/test/src/default_value_input.dart

```diff
@@ -30,4 +30,36 @@ class DefaultWithType {
   DefaultWithType();
 }
 
-// TODO: const, non-literal (const Duration, etc)
+@JsonSerializable()
+class DefaultWithConstObject {
+  @JsonKey(defaultValue: const Duration())
+  Object field;
+
+  DefaultWithConstObject();
+}
+
+enum Enum { value }
+
+@JsonSerializable()
+class DefaultWithNestedEnum {
+  @JsonKey(defaultValue: [Enum.value])
```

Comment:
nit: don't you still need the `const` here?

Reply:
you do if you want the code to run in a Dart 1 VM

Reply:
Tests pass fine in w/ our current setup @ latest Dart VM...